### ISSUE TITLE
[CI] GeoGridAggAndQueryConsistencyIT.testGeoShapeGeoHex failure

### ISSUE THREAD

**Issue by astefan** [elastic/elasticsearch#92551]
### CI Link

https://gradle-enterprise.elastic.co/s/j3ajt4vzqybxs

### Repro line

gradlew ':x-pack:plugin:spatial:internalClusterTest' --tests "org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT.testGeoShapeGeoHex" -Dtests.seed=67641D5B7DA4755B -Dtests.locale=en-IN -Dtests.timezone=Canada/Saskatchewan -Druntime.java=17 -Dtests.fips.enabled=true

### Does it reproduce?

No

### Applicable branches

main

### Failure excerpt

The test in CI link above doesn't repro for me but I did run the test with multiple iterations. After around 250 of them, one failed for me:

```
REPRODUCE WITH: gradlew ':x-pack:plugin:spatial:internalClusterTest' --tests "org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT.testGeoShapeGeoHex {seed=[3478FF74E8D66BB6:A2F98EBD906145BB]}" -Dtests.seed=3478FF74E8D66BB6 -Dtests.locale=sr-RS -Dtests.timezone=Europe/Luxembourg -Druntime.java=19

org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT > testGeoShapeGeoHex {seed=[3478FF74E8D66BB6:A2F98EBD906145BB]} FAILED
    java.lang.AssertionError: Expected hits at precision 15
    Expected: <3L>
         but: was <4L>
        at __randomizedtesting.SeedInfo.seed([3478FF74E8D66BB6:A2F98EBD906145BB]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT.assertQuery(GeoGridAggAndQueryConsistencyIT.java:224)
        at org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT.doTestGrid(GeoGridAggAndQueryConsistencyIT.java:215)
        at org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT.doTestGeohexGrid(GeoGridAggAndQueryConsistencyIT.java:124)
        at org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT.testGeoShapeGeoHex(GeoGridAggAndQueryConsistencyIT.java:92)
```

**Comment by craigtaverner** [comment]
Failed again today at https://gradle-enterprise.elastic.co/s/ahectnr2fbapo/tests/:x-pack:plugin:spatial:internalClusterTest/org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT/testGeoShapeGeoHex?top-execution=1

    ./gradlew ':x-pack:plugin:spatial:internalClusterTest' --tests "org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT.testGeoShapeGeoHex" -Dtests.seed=377E930DF9FCC337 -Dtests.locale=tr -Dtests.timezone=Asia/Tomsk -Druntime.java=17

**Comment by craigtaverner** [comment]
Also fails with `-Dtests.seed=377E930DF9FCC337`

**Comment by craigtaverner** [comment]
Both the above failures occured at the max precision of 15, so I suspected a rounding error, however, when running with `Dtests.iters=1000` we see a bit more variation in failures. We saw 7/535 failures (timed out so we did not run the full 1000). Including the above two failures:

    Precision  Seed
    15              -Dtests.seed=67641D5B7DA4755B
    15              -Dtests.seed=377E930DF9FCC337
    12              -Dtests.seed=3FEC81056CB34D61:47B737999D68B0DD
    14              -Dtests.seed=3FEC81056CB34D61:85473593DD01778F
    15              -Dtests.seed=3FEC81056CB34D61:773CBFAB9B5DEB8
    13              -Dtests.seed=3FEC81056CB34D61:A7DDADC9BD45EE70
    15              -Dtests.seed=3FEC81056CB34D61:6B4D4B687302DF35
    15              -Dtests.seed=3FEC81056CB34D61:D69CAC5ED398F0FC
    4               -Dtests.seed=3FEC81056CB34D61:7561946166B6AF96

The last one shows that the failure can occur even at low precison. We'll focus on that case for debugging.

**Comment by craigtaverner** [comment]
After an initial fix in #92551, running with `-Dtests.seed=3FEC81056CB34D61:7561946166B6AF96` failed, so the fix is not covering all cases. It could be that only the targetted case was fixed. So I ran with `-Dtests.iters=1000` and 2/866 failed, which is a better rate than before.

```
Precision  Seed
2   -Dtests.seed=CB74913019F503A4:549B1E0BBEE26DD2
3   -Dtests.seed=CB74913019F503A4:99836647C6CD1288
```

**Comment by craigtaverner** [comment]
Having re-tested all previous failures, it seems only the three failures at low precision remain, so the initial PR does fix all 8 others. Also it seems that with fixed tests we had the aggregation missing values that the query found, while with the remaining failures we have fewer query results than aggregation results.

```
Precision. Query/Agg    Seed
4         4/3     -Dtests.seed=3FEC81056CB34D61:7561946166B6AF96
2          105/106 -Dtests.seed=CB74913019F503A4:549B1E0BBEE26DD2
2          ?/? -Dtests.seed=CB74913019F503A4:99836647C6CD1288
```
